components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -2106,4 +2282,176 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
+
+        List<String> allowedSignatureAlgorithms = IdentityUtil
+                .getPropertyAsList(FAPI_SIGNATURE_ALGORITHM_CONFIGURATION);
+        if (allowedSignatureAlgorithms.isEmpty()) {
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.ES256);
+            allowedSignatureAlgorithms.add(OAuthConstants.SignatureAlgorithms.PS256);
+        }
+        if (signatureAlgorithm != null && !allowedSignatureAlgorithms.contains(signatureAlgorithm)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid signature algorithm requested");
+        }
+    }
+
+    /**
+     * FAPI validation to restrict the encryption algorithms.
+     * Link - https://openid.net/specs/openid-financial-api-part-2-1_0.html#encryption-algorithm-considerations
+     * @param encryptionAlgorithm
+     * @throws IdentityOAuthClientException
+     */
+    private void filterAllowedFAPIEncryptionAlgorithms(String encryptionAlgorithm)
+            throws IdentityOAuthClientException {
+
+        if (encryptionAlgorithm.equals(OAuthConstants.RESTRICTED_ENCRYPTION_ALGORITHM)) {
+            throw handleClientError(INVALID_REQUEST, "Invalid encryption algorithm requested");
+        }
+    }
+
+
+    /**
+     * If there are multiple hostnames in the registered redirect_uris,
+     * the Client MUST register a sector_identifier_uri.
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     * @param redirectURIs list of callback urls sent in the request
+     * @throws IdentityOAuthClientException
+     */
+    private void validateRedirectURIForPPID(List<String> redirectURIs) throws IdentityOAuthClientException {
+
+        if (redirectURIs.size() > 1) {
+            String hostname = URI.create(redirectURIs.get(0)).getHost();
+            for (String redirectURI : redirectURIs) {
+                URI uri = URI.create(redirectURI);
+                if (uri != null && !uri.getHost().equals(hostname)) {
+                    throw handleClientError(INVALID_REQUEST, "Sector identifier URI is needed for PPID calculation");
+                }
+            }
+        }
+    }
+
+    /**
+     * The value of the sector_identifier_uri MUST be a URL using the https scheme
+     * The values of the registered redirect_uris MUST be included in the elements of the array.
+     * https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+     *
+     * @param sectorIdentifierURI sector identifier URI
+     * @param redirectURIs        callBack URLs
+     * @throws IdentityOAuthClientException
+     */
+    private void validateSectorIdentifierURI(String sectorIdentifierURI, List<String> redirectURIs) throws
+            IdentityOAuthClientException {
+
+        if (StringUtils.isNotBlank(sectorIdentifierURI)) {
+            URI uri = URI.create(sectorIdentifierURI);
+            if (uri != null) {
+                String scheme = uri.getScheme();
+                if (StringUtils.isBlank(scheme) || !scheme.equals("https")) {
```

Comment:
Make https a constant (reuse if a constant already exists). 
schemes are case insensitive https://www.rfc-editor.org/rfc/rfc3986#section-3.1

```suggestion
                if (!"https".equalsIgnoreCase(scheme)) {
```

Reply:
changed in https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2178/commits/feb4731bf60075ddf78c6651b076e544ceea8add